### DISCUSSION TITLE
Инжекция зависимостей

### DISCUSSION THREAD

**Discussion by pingvin1308** [cleannetcode/LessonMonitor#86] [ASP.NET Core]
Для выполнения домашней работы, вам необходимо открыть и сделать форк основного репозитория по ссылке:
https://github.com/cleannetcode/LessonMonitor/tree/develop

По выполнению работы, вам необходимо создать пулл реквест из вашей ветки с выполненной домашкой в основной репозиторий.

Почитать:
https://metanit.com/sharp/aspnet5/6.2.php
https://docs.microsoft.com/ru-ru/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-5.0

Сделать:
Реализовать использование DI в LessonMonitor.
К примеру указать зависимость для UsersController -> UsersService -> UsersRepository

Проверить:
провести ревью кода любого чужого пулл реквеста относящегося к данной домашке.

**Comment by eniluck** [comment]
Вот ещё вариант объяснения разницы AddTransient, AddScoped, AddSingleton со [stackoverflow](https://stackoverflow.com/questions/38138100/addtransient-addscoped-and-addsingleton-services-differences).
